src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -131,15 +133,26 @@ default String getName() {
      * @param name Name of asset
      * @return Asset if present, empty otherwise
      */
+    @Deprecated
     default Optional<Asset> getAsset(String name) {
         return Sponge.getAssetManager().getAsset(this, name);
     }
 
+    /**
+     * Retrieves the {@link PackInfo} owned by this plugin from the
+     * {@link PackRepository}.
+     *
+     * @return The plugin's resource pack.
+     */
+    default PackInfo getPack() {
+        return Sponge.getServer().getPackRepository().getPack(this);
```

Comment:
As far as I know, resource packs and data packs have different information, even as they can share the same pack mcmeta file. This prevents a plugin from having both packs, which is the case for forge mods, etc.

Reply:
This will probably have to wait for a full impl until the client api gets fleshed out a bit more. As it is now, `ResourceTypes.ASSETS` is restricted to the client, and `DATA` to the server.

Reply:
I suggest removing this method. Plugin Containers may exist without the server. This method itself is default, meaning that the removal will not reduce functionality.

Reply:
Instead of removing this method, we can leave it abstract so that implementations are correct.

Reply:
We should consider moving this to the client and server interfaces instead.